lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2362,6 +2362,200 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
+        gridRank, utils::IteratorType::parallel);
+    SmallVector<int64_t> resultShape = makeShapeTorchCompatible(inputShape);
+    resultShape[2] = gridShape[1];
+    resultShape[3] = gridShape[2];
+    Value resultFinal =
+        rewriter.create<tensor::EmptyOp>(loc, resultShape, floatType);
+    auto resultFinalType = resultFinal.getType().cast<RankedTensorType>();
+    Value sGrid =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, resultFinalType,
+                ValueRange{gridCollapsed_0, gridCollapsed_1}, resultFinal,
+                gridMaps, gridIterators,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value gr_0 = args[0];
+                  Value gr_1 = args[1];
+                  Value oneFloat = b.create<arith::ConstantOp>(
+                      loc, b.getFloatAttr(gridElementType, 1.0));
+                  Value gplus_0 = b.create<arith::AddFOp>(loc, gr_0, oneFloat);
+                  Value gplus_1 = b.create<arith::AddFOp>(loc, gr_1, oneFloat);
+                  Value result_0 =
+                      b.create<arith::MulFOp>(loc, gplus_0, innerDim_0e);
+                  Value result_1 =
+                      b.create<arith::MulFOp>(loc, gplus_1, innerDim_1e);
+                  Value lower_0 =
+                      b.create<arith::FPToSIOp>(loc, int64type, result_0);
+                  Value lower_1 =
+                      b.create<arith::FPToSIOp>(loc, int64type, result_1);
+                  Value oneInt = b.create<arith::ConstantOp>(
+                      loc, b.getIntegerAttr(int64type, 1));
+                  Value upper_0 =
+                      b.create<arith::AddIOp>(loc, int64type, lower_0, oneInt);
+                  Value upper_1 =
+                      b.create<arith::AddIOp>(loc, int64type, lower_1, oneInt);
+                  Value notValid_0 = createGreaterThan(b, loc, int64type,
+                                                       upper_0, innerDim_0c);
+                  Value notValid_1 = createGreaterThan(b, loc, int64type,
+                                                       upper_1, innerDim_1c);
+                  Value upperValid_0 = b.create<arith::SelectOp>(
+                      loc, notValid_0, lower_0, upper_0);
+                  Value upperValid_1 = b.create<arith::SelectOp>(
+                      loc, notValid_1, lower_1, upper_1);
+                  Value lw_0 = b.create<arith::IndexCastOp>(
+                      loc, b.getIndexType(), lower_0);
+                  Value lw_1 = b.create<arith::IndexCastOp>(
+                      loc, b.getIndexType(), lower_1);
+                  Value up_0 = b.create<arith::IndexCastOp>(
+                      loc, b.getIndexType(), upperValid_0);
+                  Value up_1 = b.create<arith::IndexCastOp>(
+                      loc, b.getIndexType(), upperValid_1);
+                  Value N = b.create<linalg::IndexOp>(loc, 0);
```

Comment:
Move these to the top of the linalg.generic body. Anything constant like is good to declare at the beginning.